background.ts

```diff
@@ -388,25 +389,34 @@ Electron.ipcMain.on('k8s-state', (event) => {
   event.returnValue = k8smanager.state;
 });
 
-Electron.ipcMain.on('k8s-reset', async(event, arg) => {
-  try {
-    // If not in a place to restart than skip it
-    if (![K8s.State.STARTED, K8s.State.STOPPED, K8s.State.ERROR].includes(k8smanager.state)) {
-      console.log(`Skipping reset, invalid state ${ k8smanager.state }`);
+Electron.ipcMain.handle('current-port', event => k8smanager.port);
 
-      return;
-    }
+Electron.ipcMain.on('k8s-reset', async(_, arg) => {
+  await doK8sReset(arg);
+});
+
+async function doK8sReset(arg = ''): Promise<void> {
+  // If not in a place to restart than skip it
+  if (![K8s.State.STARTED, K8s.State.STOPPED, K8s.State.ERROR].includes(k8smanager.state)) {
+    console.log(`Skipping reset, invalid state ${ k8smanager.state }`);
 
-    if (k8smanager.version !== cfg.kubernetes.version ||
-      (await k8smanager.cpus) !== cfg.kubernetes.numberCPUs ||
-      (await k8smanager.memory) !== cfg.kubernetes.memoryInGB * 1024) {
+    return;
+  }
+
+  try {
+    if (['slow', 'fast'].includes(arg)) {
+      // Leave arg as is
+    } else if ((k8smanager.version !== cfg.kubernetes.version ||
+        (await k8smanager.cpus) !== cfg.kubernetes.numberCPUs ||
+        (await k8smanager.memory) !== cfg.kubernetes.memoryInGB * 1024 ||
+        (k8smanager.port) !== cfg.kubernetes.port)) {
```

Comment:
Hmm, this means that `arg` can still be arbitrary other values (e.g. `''`, `'whatever'`) and we never guard against it.  That's an existing issue, though.

Reply:
Yes on a separate issue. It looks like the arg was originally intended to be used in an error message, but wasn't needed, and the `arg` variable was repurposed.